restartmanager/restartmanager.go

```diff
@@ -90,7 +91,7 @@ func (rm *restartManager) ShouldRestart(exitCode uint32, hasBeenManuallyStopped
 	case rm.policy.IsOnFailure():
 		// the default value of 0 for MaximumRetryCount means that we will not enforce a maximum count
 		if max := rm.policy.MaximumRetryCount; max == 0 || rm.restartCount < max {
-			restart = exitCode != 0
+			restart = (exitCode != 0) || (health.Status == types.Unhealthy)
```

Comment:
Does `health.Status == types.Unhealthy` happen only when Unhealthy policy is on?

Reply:
This happens only on the `IsOnFailure` policy, I haven't defined a special policy for healthchecks to keep it aligned with docker